src/memory/memoryManager.ts

```diff
@@ -0,0 +1,28 @@
+import { Memory } from './memory';
+import { MaybeRelocatable, Relocatable } from './primitives/relocatable';
+import { Uint, UnsignedInteger } from './primitives/uint';
+
+export class MemorySegmentManager {
+  segmentSizes: Map<Uint, Uint>;
+  memory: Memory;
+
+  constructor() {
+    this.segmentSizes = new Map();
+    this.memory = new Memory();
+  }
+
+  addSegment(): Relocatable {
+    const ptr = new Relocatable(this.memory.numSegments, 0);
+    this.memory.numSegments = UnsignedInteger.toUint(
+      this.memory.numSegments + 1
+    );
+    return ptr;
+  }
+
+  loadData(address: Relocatable, data: MaybeRelocatable[]): Relocatable {
+    data.forEach((d, index) =>
```

Comment:
i had to modify the forEach into a for loop to accomodate result based

Reply:
let's merge your PR first then, I will rebase